mflowgen/common/power-domains/outputs/pe-add-endcaps-welltaps-setup.tcl

```diff
@@ -9,31 +8,53 @@
 # locations/orientations, which the foundation flow does not natively support
 
 if {[expr {$ADK_END_CAP_CELL == ""} && {$ADK_WELL_TAP_CELL == ""}]} {
-    
+
     deleteInst *TAP*
     deleteInst *ENDCAP*
-    deleteInst *tap* 
+    deleteInst *tap*
+
+    # Align tap cells with M3 pitch so that the M1 VPP pin is directly aligned with the M3 VDD net
+
+    # Get M3 min width and signal routing pitch as defined in the LEF
 
-    set flatten_effort $::env(flatten_effort) 
-    set edge_offset    12.91
+    set M3_min_width    [dbGet [dbGetLayerByZ 3].minWidth]
+    set M3_route_pitchX [dbGet [dbGetLayerByZ 3].pitchX]
 
-    addInst -cell BOUNDARY_PTAPBWP16P90_VPP_VSS -inst top_endcap_tap_1
-    addInst -cell BOUNDARY_PTAPBWP16P90_VPP_VSS -inst top_endcap_tap_2
+    # Set M3 stripe variables
+
+    set M3_str_width            [expr  3 * $M3_min_width]
+    set M3_str_pitch            [expr 10 * $M3_route_pitchX]
+
+    set M3_str_intraset_spacing [expr ($M3_str_pitch - 2*$M3_str_width)/2]
+    set M3_str_interset_pitch   [expr 2*$M3_str_pitch]
+    set M3_str_offset           [expr $M3_str_pitch + $M3_route_pitchX/2 - $M3_str_width/2]
```

Comment:
Later we should probably refactor this such that there is a single source of truth about the power stripe parameters, but this is ok for now to get this tapeout out the door.